test/e2e/antreapolicy_test.go

```diff
@@ -3320,6 +3320,48 @@ func testFQDNPolicyInClusterService(t *testing.T) {
 	failOnError(k8sUtils.DeleteACNP(builder.Name), t)
 }
 
+// testFQDNPolicyTCP
+func testFQDNPolicyTCP(t *testing.T) {
+	// The ipv6-only test env doesn't have IPv6 access to the web.
+	skipIfNotIPv4Cluster(t)
+	// It is convenient to have higher log verbosity for FQDNtests for troubleshooting failures.
+	logLevel := log.GetLevel()
+	log.SetLevel(log.TraceLevel)
+	defer log.SetLevel(logLevel)
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("test-acnp-fqdn-tcp").
+		SetTier("application").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{NSSelector: map[string]string{}}})
+	builder.AddFQDNRule("github.com", ProtocolTCP, nil, nil, nil, "", nil, crdv1alpha1.RuleActionDrop)
+
+	testcases := []podToAddrTestStep{
+		{
+			Pod(namespaces["y"] + "/a"),
+			"github.com",
+			80,
+			Dropped,
+		},
+	}
+	acnp, err := k8sUtils.CreateOrUpdateACNP(builder.Get())
+	failOnError(err, t)
+	failOnError(waitForResourceReady(t, timeout, acnp), t)
+	for _, tc := range testcases {
+		destIP := k8sUtils.digDNS(tc.clientPod.PodName(), tc.clientPod.Namespace(), tc.destAddr, true)
+		log.Tracef("Probing: %s -> %s(%s)", tc.clientPod.PodName(), tc.destAddr, destIP)
+		connectivity, err := k8sUtils.ProbeAddr(tc.clientPod.Namespace(), "pod", tc.clientPod.PodName(), destIP, tc.destPort, ProtocolTCP)
```

Comment:
So the point of the test is that it proactively queries the domain using TCP to make the datapath learn the IP, then query the IP directly to check if it's learned?

Reply:
Yes exactly.